.github/workflows/yateto-ci.yml

```diff
@@ -0,0 +1,70 @@
+name: Yateto CI
+
+on: push
+
+jobs:
+  general:
+    runs-on: ubuntu-latest
+    container: 
+      image: ravilmobile/yateto-env:latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install Yateto
+      run: |
+        pip3 install -e .
+    
+    - name: Python Tests
+      run: |
+        python3 -m unittest tests/internals/*.py
+        
+    - name: Interface Tests
+      run: |
+        . /etc/profile.d/z10_spack_environment.sh && . /etc/profile.d/z20_additional_env.sh
+        cd ./tests/interface
+        for build_type in Debug Release; do
+          mkdir -p ./build-${build_type} && cd ./build-${build_type}
+          cmake .. -DCMAKE_BUILD_TYPE=${build_type}
+          make
+          make test
+          cd ..
+        done
+
+  codegen:
+    runs-on: ubuntu-latest
+    container: 
+      image: ravilmobile/yateto-env:latest
+    env:
+      CTEST_OUTPUT_ON_FAILURE: 1
+    strategy:
+      matrix:
+        generator: [Eigen, LIBXSMM, OpenBLAS]
+        
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install Yateto
+      run: |
+        pip3 install -e .
+
+    - name: Codegen Tests
+      run: |
+        . /etc/profile.d/z10_spack_environment.sh && . /etc/profile.d/z20_additional_env.sh
+        cd ./tests/code-gen
+        for example in matmul minimal; do
+          for build_type in Debug Release; do
+            for precision in single double; do
+              echo " ====== Test Config: ======"
+              echo " Build: ${build_type}"
+              echo " Precision: ${precision}"
+              echo " Example: ${example}"
+              echo "==========================="
+              mkdir -p ./build-${example}-${build_type}-${precision}
+              cd ./build-${example}-${build_type}-${precision}
+              cmake .. -DEXAMPLES=${example} -DCMAKE_BUILD_TYPE=${build_type} -DPRECISION=${precision} -DVARIANT=${{ matrix.generator }} -DARCH=snb
+              make
```

Comment:
Should be make -j2 I think:
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

Reply:
Thanks, I'll fix it

> Only found one small thing (maybe). Not sure what the tests are doing but it looks reasonable.

Most of the tests are extracted from `examples`. In short, yateto generates 2 source codes for the same tensor expressions: 1)  the one which uses a back-end i.e. `libxsmm` and 2) the one which is base on nested for-loops i.e. considered to be a correct implementation. Yateto also generates a piece of code which runs both implementations and compare numerical results.

I added some `python` tests to `yateto/tests/internals` a while a go (2 tests to be honest). They  are also included in the workflow. I imagine that if somebody works with yateto and understands how some particular parts work then he/she can create and add a test, and include it in `internals`. Maybe it will become some sort of documentation at some point.

> Great that you're doing this. Maybe we also should switch to github actions for the SeisSol main repo

Thanks. I like that it was easy to specify which docker image the workflow should use for tests. Probably it is also possible for Azure. 

The next enhancement  is going to be triggering a docker build and push when somebody changes `spack.yaml` file

Reply:
Re-based this PR using the head of `SeisSol/yateto master`

Reply:
It should be possible for Azure. Imo we should switch to github actions, because it's better integrated than Azure. When I did the Azure thing, github actions was not available. Not sure whether it supports the same feature set though.

P.S.: Isn't not being able to compare tests in parallel a bug that should be caught by the CI? ;)

Reply:
In theory)))